File: python/2b.py
# I think I accidentally solved this one when trying to do part 1

with open("C:/Users/saanv/Downloads/input.txt", "r") as line:
  ids = line.readline().strip().split(",")

sum = 0

for id in ids:
  start = int(id[:id.find("-")])
  stop = int(id[id.find("-") + 1:])
  for i in range(start, stop + 1): # iterate through each number in the ranges
    for j in range (1, len(str(i))//2 + 1): # test each possible chunk size
      if str(i) == str(i)[0:j] * (len(str(i))//j):
        sum += i
        break # prevent duplicates

print(sum)

# test 1: 30260171216, took a long time but we ball, NO WAY I GOT IT FIRST TRY

File: java/2b.java
// this actually needs to be named b2.java, but I like having my files nice and neat
import java.io.*;
import java.util.*;

public class b2 {
  public static void main(String[] args) throws FileNotFoundException {
    long sum = 0;
    
    InputStream lines = new FileInputStream("C:/Users/saanv/Downloads/input.txt");
    Scanner input = new Scanner(lines);
    String[] ids = input.nextLine().strip().split(",");
    for (String id : ids) {
      long start = Long.parseLong(id.substring(0, id.indexOf("-")));
      long stop = Long.parseLong(id.substring(id.indexOf("-") + 1));
      for (Long i = start; i <= stop; i++) {
        for (int j = 1; j <= Long.toString(i).length() / 2; j++) {
          if (Long.toString(i).equals((Long.toString(i)).substring(0, j).repeat(Long.toString(i).length() / j))) {
            sum += i;
            break;
          }
        }
      }
    }
    System.out.print(sum);
  }
}
